exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/AsyncPageReader.java

```diff
@@ -74,12 +82,17 @@ protected void loadDictionaryIfExists(final ColumnReader<?> parentStatus,
         handleAndThrowException(e, "Error Reading dictionary page.");
       }
       // parent constructor may call this method before the thread pool is set.
```

Comment:
Why not ensure parent calls the c'tor with pool != null?

That way all variables can be final and non-null, the `offer(...)` is only done once in the c'tor, and the internal classes can directly access the `pageQueue`.